File: ios/sources/commands/ARESLineWidthCommand.m
//
//  ARESLineWidthCommand.m
//  ares
//
//  Created by PonyCui on 2018/11/29.
//  Copyright © 2018年 XT Studio. All rights reserved.
//

#import "ARESLineWidthCommand.h"
#import "ARESView.h"

@implementation ARESLineWidthCommand

- (instancetype)initWithLineWidth:(CGFloat)lineWidth {
    self = [super init];
    if (self) {
        _lineWidth = lineWidth;
    }
    return self;
}

- (void)draw:(CGContextRef)ctx view:(ARESView *)view {
    CGContextSetLineWidth(ctx, self.lineWidth);
    view.currentState.lineWidth = self.lineWidth;
}

@end

File: android/ares/src/main/java/com/xt/ares/commands/ARESLineWidthCommand.kt
package com.xt.ares.commands

import android.graphics.Canvas
import com.xt.ares.ARESCommand
import com.xt.ares.ARESView

class ARESLineWidthCommand(val value: Double): ARESCommand() {

    override fun draw(view: ARESView, canvas: Canvas) {
        super.draw(view, canvas)
        view.currentPaint.strokeWidth = value.toFloat()
    }

}